### PR TITLE
create loss object and loss_type.csv

### PR DESCRIPTION
**Pull request by odscjen**
**Related issues**

closes https://github.com/GFDRR/rdl-standard/issues/126

Updates Loss component.

Removes `time_start`, `time_end` and `time_year` as they are all covered in the top level `temporal` object.

Needs the `Vulnerability` PR (https://github.com/GFDRR/rdl-standard/pull/124) to be merged first as it contains the `Impact` object referenced by `Loss`.

**Merge checklist**

- [x] Update the changelog ([style guide](developer_docs.md#changelog-style-guide))
- [x] Run `./manage.py` pre-commit

If you added or removed a field:

- [ ] Update the `collapse` option of the jsonschema directives for dataset, resource, hazard, exposure, vulnerability and loss on `reference/schema.md`

Re. this 3rd item, @duncandewhurst how do we do that?

### CODE REVIEW

**Comment by duncandewhurst** [comment]
> Re. this 3rd item, @duncandewhurst how do we do that?

You need to edit the relevant lines of `schema.md`, for example, for loss:

https://github.com/GFDRR/rdl-standard/blob/71296b83b7f087b575f4c4e271826a72d28aec3e/docs/reference/schema.md?plain=1#L390

The [`collapse` option](https://sphinxcontrib-opendataservices-jsonschema.readthedocs.io/en/latest/use.html#option-collapse) accepts a comma-separated list of keys to collapse in the schema table generated by the directive.

In order to keep the table lengths manageable generally, we want to collapse keys that reference a definition since each definition has its own schema table under the sub-schemas heading on the reference page.

So for loss, you would collapse `cost` and `impact` since they reference the `Cost` and `Impact` definitions respectively, e.g.

```md
collapse: cost,impact
```

File: schema/rdl_schema_0.1.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "risk_data_type": {
       "title": "Risk data type",
       "type": "string",
-      "description": "The type of risk data included in the dataset, from the closed [risk data type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/reference/codelists/#risk-data-type).",
+      "description": "The type of risk data included in the dataset, from the closed [risk data type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/references/codelists/#risk-data-type).",
```
Comment (duncandewhurst) [inline schema/rdl_schema_0.1.json:39]: Not sure why all the codelist links have been updated, the old links (without the trailing 's' on reference) were correct.

```suggestion
      "description": "The type of risk data included in the dataset, from the closed [risk data type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/reference/codelists/#risk-data-type).",
```

File: schema/rdl_schema_0.1.json [hunk 2]
```diff
@@ -1796,7 +1678,164 @@
...
+            "Buildings",
+            "Infrastructures",
+            "Population",
+            "Natural environment"
+          ]
+        },
+        "cost": {
+          "title": "Asset cost",
+          "type": "array",
+          "description": "The exposure costs associated with specific elements of assets detailed in the dataset.",
```
Comment (duncandewhurst) [inline schema/rdl_schema_0.1.json:1770]: Is 'exposure costs' the correct term here? If I understood correctly, loss data is derived from a combination of hazard, exposure and vulnerability data, i.e. the cost (value) of assets that are exposed to a hazard could be greater than the cost (value) of assets that are actually lost, depending on the vulnerability function that is applied. As such, I think the description should be something more like:

```suggestion
          "description": "The costs associated with the loss of specific elements of assets detailed in the dataset.",
```

**Review by duncandewhurst — CHANGES_REQUESTED** [review]

File: schema/rdl_schema_0.1.json (hunk 1 above)
Comment (odscjen) [reply schema/rdl_schema_0.1.json:39]: that was my mistake!

**Comment by odscjen** [comment]
I think I checked the right bit of the collapse and it had all the fields listed, not sure if that was correct...

File: schema/rdl_schema_0.1.json [hunk 3]
```diff
@@ -2127,7 +2010,164 @@
...
+            "wildfire",
+            "extratropical_cyclone",
+            "tropical_cyclone"
+          ]
+        },
+        "category": {
+          "title": "Exposure category",
+          "description": "The category of the exposed assets, from the closed [exposure category codelist](https://rdl-standard.readthedocs.io/en/{{version}}/reference/codelists/#exposure_category).",
+          "type": "string",
+          "codelist": "exposure_category.csv",
```
Comment (stufraser1) [inline schema/rdl_schema_0.1.json:2089]: Use lower case codes and _ in spaces per process_type, for example?

File: schema/rdl_schema_0.1.json
Comment (stufraser1) [inline schema/rdl_schema_0.1.json:1]: Do we need to update all enum lists to lower case codes as used in codelists?

File: schema/rdl_schema_0.1.json [hunk 4]
```diff
@@ -2127,7 +2010,164 @@
...
+          "$ref": "#/$defs/Impact"
+        },
+        "type": {
+          "title": "Loss type",
+          "type": "string",
+          "description": "The type of loss described in the dataset, from the closed [loss type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/reference/codelists/#loss_type).",
+          "codelist": "loss_type.csv",
+          "openCodelist": false,
+          "enum": [
+            "ground-up",
```
Comment (stufraser1) [inline schema/rdl_schema_0.1.json:2122]: Reflect expanded loss_type codelist

File: schema/rdl_schema_0.1.json
Comment (stufraser1) [inline schema/rdl_schema_0.1.json:1]: Similar to the hazard component, where we can have one or more events in one or a few resource files, in this loss component we can have
-  a file describing a summary of losses in a table, or a zipped folder of tables for example csv or xlsx
- Loss maps in zipped shp format
- csv file giving the loss per event in a table of 10,000 events, which is similar in size to the event set table contained in hazard component.
- do we account for being able to identify these resource types easily? Is it sufficient to be reflected in title and description of the dataset, or should we create a specific descriptor field I  loss component with codelist?

**Comment by duncandewhurst** [comment]
The `enum` for `Loss.category` needs to be updated, per https://github.com/GFDRR/rdl-standard/pull/124#issuecomment-1626343055

File: schema/rdl_schema_0.1.json
Comment (duncandewhurst) [reply schema/rdl_schema_0.1.json:1]: Yep, the `enum`s should match the CSV codelists. I suggest that we keep the scope of this PR to the loss section and then pick up any outstanding `enum` updates as part of https://github.com/GFDRR/rdl-standard/issues/129 (specifically, https://github.com/GFDRR/rdl-standard/pull/127).

Comment (duncandewhurst) [reply schema/rdl_schema_0.1.json:1]: I think that information is mostly covered by the top-level `.title` and `.description` fields and the properties of `Resource` (e.g. `.title`, `.description`, `.media_type` and `.format`). What loss-specific information would you envisage being published in `Loss.description`?

**Review by duncandewhurst — DISMISSED** [review]

**Comment by duncandewhurst** [comment]
My approval is subject to @stufraser1's comments being resolved to his satisfaction :-)

File: schema/rdl_schema_0.1.json
Comment (stufraser1) [reply schema/rdl_schema_0.1.json:1]: Loss.description would be used to describe the main hazard, exposure, aggregation level, and metric usually. E.g., 
"Flood loss estimates for residential buildings in Ghana at district level"

Comment (odscjen) [reply schema/rdl_schema_0.1.json:1]: In that example I think most of those factors are already covered by existing fields, "Flood" = `hazard_type`, "loss estimates and number of people displaced" = `loss.impact`, "Ghana" = `spatial`, "district level" = `spatial.scale`. The only one that isn't is "residential buildings" (although "buildings" is covered by `loss.category`). I don't see the need for a `loss.description` in addition to `description` and `resource.description`.

So in answer to your question
> Is it sufficient to be reflected in title and description of the dataset?

Yes

**Review by stufraser1 — APPROVED** [review]

**Pull request merged**